cryptoinvestor/views/assets.py

```diff
@@ -42,10 +42,36 @@ def get_objects(self):
                 }
             }
 
+        toasts = []
+        for asset in assets:
+            id_ = asset.get('id', 'N/A')
+            current_rate = asset.get('rate', 0.00)
+            try:
+                previous_rate = data.get(id_, {}).get(base, [])[-2].get('rate', 0.00)
+                value = round((current_rate - previous_rate), 4)
+            except IndexError:
+                value = 0
+
+            if value > 0:
+                message = f"From last time {id_} went up by {value}!"
+                color = 'green'
+            elif value < 0:
+                message = f"From last time {id_} went down by {value}!"
+                color = 'red'
+            else:
+                message = f"From last time {value} did not change!"
+                color = 'blue'
+
+            toasts.append({
+                'message': message,
+                'color': color
+            })
```

Comment:
Please export this to a separate function whose sole purpose is to generate toast messages from assets. This function is getting too complex.